### PR TITLE
CPO-503 Dockerfile build pipeline

### PR DESCRIPTION
**Pull request by tom-mckay**
https://procoretech.atlassian.net/browse/CPO-503
- Adds a build and push pipeline for the docker image.

### CODE REVIEW

**Comment by tom-mckay** [comment]
TODO - Change target branch before merging!

File: .circleci/config.yml [hunk 1]
```diff
@@ -42,6 +42,30 @@ jobs:
...
+    working_directory: ~/atlantis-helm-chart
+    steps:
+      - setup_remote_docker
+      - checkout
+      - run:
+          name: Build Image
+          command: |
+            docker build \
+              --pull \
+              -t quay.io/procoredevops/atlantis-helm-chart:$CIRCLE_TAG ~/atlantis-helm-chart/procore-atlantis/
```
Comment (brandonpollard) [inline .circleci/config.yml:57]: Do you have to do any cleanup after we build?

**Review by brandonpollard — APPROVED** [review]
lgtm

**Review by rarkebauer — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (tom-mckay) [reply .circleci/config.yml:57]: Nope. The build is all done in a remote docker environment. The built image is cleaned up automatically.

**Comment by tom-mckay** [comment]
GitHub automatically updated the target branch of this PR to main, nice!

**Pull request merged**